support-workers/src/main/scala/com/gu/support/workers/lambdas/CreateZuoraSubscription.scala

```diff
@@ -286,13 +292,14 @@ object DigitalSubscriptionGiftRedemption {
         getSubscriptionState(redemptionQueryResponse, state.requestId.toString) match {
           case Unredeemed(subscriptionId) => redeemInZuora(subscriptionId, state, redemptionData, requestInfo, zuoraService, catalogService)
           case RedeemedInThisRequest => Future.fromTry(buildHandlerResult(UpdateRedemptionDataResponse(true), state, redemptionData, requestInfo))
-          case Redeemed => Future.failed(new RuntimeException(GetCodeStatus.CodeAlreadyUsed.clientCode))
-          case NotFound => Future.failed(new RuntimeException(GetCodeStatus.NoSuchCode.clientCode))
+          case otherState: SubscriptionRedemptionState => Future.failed(new RuntimeException(otherState.clientCode))
         }
     )
 
-  private def getSubscriptionState(existingSub: SubscriptionRedemptionQueryResponse, requestId: String) =
+  def getSubscriptionState(existingSub: SubscriptionRedemptionQueryResponse, requestId: String) =
     existingSub.records match {
+      case existingSubFields :: Nil if existingSubFields.contractEffectiveDate.plusYears(1).isBefore(LocalDate.now()) =>
```

Comment:
to aid testability, I would prefer to pass today's date in, keep the non ref transparent stuff at the outside.  I realise it would add to the function signatures so it depends if you think that's a valid use of boilerplate though?

Reply:
I'd rather not unless we specifically need to for testing purposes as it just adds to the cognitive load when reading through the code

Reply:
fair enough, I know what you mean 👍 

> as it just adds to the cognitive load when reading through the code

although that is the referential transparency/FP simplicity debate in a nutshell!